RFC.md

```diff
@@ -0,0 +1,393 @@
+dump out values in more readable forms.
+
+### Packed Representation
+
+The Packed Representation is a compact representation optimized
+for transmitting, storing, displaying, and using multihashes.
+It is optimized to avoid wasting space. It is also tuned to be
+incorporated into other values, such as content-addressed store
+identifiers, URI/URLs, unix filesystem paths, and more. This is
+the primary representation of a multihash format.
+
+The Packed Representation is structured as follows:
+
+    <hash-function-code-v><digest-length-v><digest-value-bin>
+
+Where:
+- `<hash-function-code-v>` is a varint storing the code of the hash function
+  according to a pre-determined multihash table, and the Standard Multihash
+  Table.
+- `<digest-length-v>` is a varint storing the length of `<digest-value>`,
+  in bytes.
+- `<digest-value-bin>` is the hash function digest value, binary-packed.
+
+#### Hash Function Code
+
+The Hash Function Code in the packed representation is an unsigned
+varint, following the format specified in section (TODO). This code
+references a pre-determined multihash table, usually the standard
+Standard Multihash Table maintained alongside this specification.
+
+It is possible for users to define and use their own tables, which
+MUST be compatible with the Standard Multihash Table, by using
+only the code ranges left undefined for this purpose. Section (TODO)
+explains the Standard Multihash Table and which ranges can be used.
+
+#### Digest Length in Bytes
+
+The `<digest-length>` counts bytes, not bits. This reduces the storage
+required for most hash function values: 256 and 512 bit lengths are
+represented as "32" and "64", using 1-byte varints instead of 2-byte
+varints.
+
+This size reduction exploits the facts that: (a) most common computer
+architectures use 8-bit words, (b) most networks transit sequences of
+bytes, (c) most storage systems store sequences of bytes, and (d) most
+if not all widely used hash functions have standard digest lengths
+divisible by 8.
+
+It is possible that some hash functions are used with digests of lengths
+not evenly divisble by 8. In such rare cases, the function implementation
+with multihash should define a byte-aligned version, usually by adding a
+pre-determined amount of padding bits at the end of the value. Such
+padding transformations can be well-defined, and would likely have to
+exist to support such hash functions in current computer architectures,
+storage systems, and networks. Therefore this specification is comfortable
+leaving this up to the implementations and users.
+
+#### Digest Value
+
+The `<digest-value-bin>` is simply a binary-packed representation of the
+hash function value. This value ensures the entire Multihash Packed
+Representation is as compact as it could be, wasting no space to
+represent the digest value.
+
+Base encoding for strings must be performed around the whole multihash
+value,
+
+### Explanation Representation
```

Comment:
The explanation representation has so far been used very rarely. Instead we almost always simply convert the digest-value to a given base.

In the begining this used to be base58 implicitly, but with IPLD and [CID](https://github.com/ipld/cid) we started using [multibase](https://github.com/multiformats/multibase), which prepends a varint signifying the base encoding of the string.